modules/wallet/signtransaction.go

```diff
@@ -0,0 +1,64 @@
+package wallet
+
+import (
+	"bytes"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+type hashSigner func(crypto.Hash, crypto.SecretKey) (crypto.Signature, error)
```

Comment:
I'm guess that you are trying to avoid a dependency on crypto.SignHash... but why? It seems like unneeded extra complexity to me.